crates/router/src/core/routing.rs

```diff
@@ -1266,16 +1279,32 @@ pub async fn retrieve_linked_routing_config(
             }
         }
 
-        // Fetch all dynamic algorithms
+        // Prepare futures for dynamic algorithms
         for algorithm_id in dynamic_algorithm_ids {
-            let record = db
-                .find_routing_algorithm_metadata_by_algorithm_id_profile_id(
-                    &algorithm_id,
-                    profile_id,
-                )
-                .await
-                .to_not_found_response(errors::ApiErrorResponse::ResourceIdNotFound)?;
-            active_algorithms.push(record.foreign_into());
+            let store = state.store.clone();
+            let profile_id_clone = profile_id.clone();
```

Comment:
```suggestion
            let profile = profile_id.clone();
```
Nit: we don't name something with a suffix or prefix considering the type, So we don't name like currency_vector, instead we can say something like currencies.